### ISSUE TITLE
Tailor Support

### ISSUE THREAD

**Issue by kombimedia** [OFFLINE-GmbH/oc-site-search-plugin#119]
Hi there,

Any plans to support October v3 and in particular Tailor generated content?

Cheers,

Cy

**Comment by tobias-kuendig** [comment]
Yes, absolutely! I just did not find the time to implement the feature yet, but it is on the roadmap.

In the meantime, you can implement it yourself in our `app/Provider.php`:

```php
    Event::listen('offline.sitesearch.query', function ($query) {
            $controller = \Cms\Classes\Controller::getController() ?? new \Cms\Classes\Controller();

            $items = \Tailor\Models\EntryRecord::inSection('Your\Section')
                ->applyPublishedStatus()
                ->where(function ($q) use ($query) {
                    $q->where('title', 'like', "%{$query}%")
                        ->orWhere('content', 'like', "%{$query}%")
                        ->orWhere('some_other_field', 'like', "%{$query}%");
                })
                ->get();

            $results = $items->map(function ($item) use ($query, $controller) {
                $relevance = mb_stripos($item->title, $query) !== false ? 2 : 1;

                return [
                    'title' => $item->title,
                    'text' => $item->content,
                    'url' => $controller->pageUrl('target-page-filename', ['slug' => $item->slug]),
                    'thumb' => optional($item->image)->first(),
                    'relevance' => $relevance,
                ];
            });

            return [
                'provider' => 'Project',
                'results' => $results,
            ];
        });

```

**Comment by kombimedia** [comment]
@tobias-kuendig

Awesome, that worked a treat!

So I guess I just need to create a listener for each blueprint I want to be searchable, all in the same provider?

Thank you so much for your help!

**Comment by tobias-kuendig** [comment]
Exactly, that should work until an official solution is provided by the plugin.

**Comment by kombimedia** [comment]
Thanks @tobias-kuendig

I have implemented search for all of my blueprints, works perfectly!

**Comment by kombimedia** [comment]
Hi @tobias-kuendig 

After pushing the custom provider to staging, the search does not seem to be picking up tailor content.

Checking a few things;

- The provider is on the staging server in the correct dir.
- Tailor content has been copied to the staging site.
- I have cleared the application cache.
- I have cleared the browser cache and cookies.

Anything else I should be checking, or any ideas as to what could be going on here. Everything works as expected in the local environment.

Cheers

**Comment by tobias-kuendig** [comment]
Did you migrate the tailor schemas? Also, check if the `Event::listen` callback is firing. I'd say the problem is either that the event listener is not firing correctly or your Tailor schema is somehow different than local and the query does not return any results because of that.

**Comment by kombimedia** [comment]
Thanks @tobias-kuendig , I got it sorted. 

`Event::listener` wasn't firing on the staging server. As the site was upgraded from v2, the `Provider.php` file wasn't loaded by the default configuration and I hadn't configured my service provider correctly. 

Working perfect now, thanks for your help!

**Comment by tobias-kuendig** [comment]
Support for Tailor content has landed in v1.7.0. 

Docs: https://github.com/OFFLINE-GmbH/oc-site-search-plugin#tailor
